### PR TITLE
Initialize `TransactionHashes` field for `GenesisBlock`

### PR DESCRIPTION
**Pull request by m-Peter**
Closes: https://github.com/onflow/flow-evm-gateway/issues/371

## Description

This needs to be always initialized as the default value for `interface{}` is `null` (https://github.com/onflow/flow-evm-gateway/blob/main/api/models.go#L281).

______

For contributor use:

- [x] Targeted PR against `master` branch
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [x] Code follows the [standards mentioned here](https://github.com/onflow/flow-nft/blob/master/CONTRIBUTING.md#styleguides).
- [x] Updated relevant documentation 
- [x] Re-reviewed `Files changed` in the Github PR explorer
- [x] Added appropriate labels 

## Summary by CodeRabbit

- **New Features**
  - Enhanced the Genesis Block structure to include transaction hashes, improving how transaction data is stored and utilized.
  
- **Impact**
  - This change may influence the logic and flow of block handling, allowing for better transaction management at the genesis level.

### CODE REVIEW

File: models/block.go [hunk 1]
```diff
@@ -20,7 +20,10 @@ var (
 	EarliestBlockNumber  = big.NewInt(0)
 )
 
-var GenesisBlock = &Block{Block: types.GenesisBlock}
+var GenesisBlock = &Block{
+	Block:             types.GenesisBlock,
+	TransactionHashes: []gethCommon.Hash{},
```
Comment (sideninja) [inline models/block.go:25]: what does this change affect then with block transactions?

**Review by sideninja — APPROVED** [review]

File: models/block.go (hunk 1 above)
Comment (m-Peter) [reply models/block.go:25]: I think this is how default values for interfaces work, in Golang. If it doesn't have a concrete value, it will assign `null`.

Comment (sideninja) [reply models/block.go:25]: I understand that, but why is it an issue? because of some serialization that it changes or?

Comment (m-Peter) [reply models/block.go:25]: Yeah, this issue here https://github.com/onflow/flow-evm-gateway/issues/371#issuecomment-2262828643. serialization related.

**Pull request merged**